### ISSUE TITLE
Build docker image as multiplatform image

### ISSUE THREAD

**Issue by csydvs** [pact-foundation/pact-ruby-cli#51]
Currently, the pact-cli can only run on Intel/AMD based architectures. With the advent of Apple's M1 chips and Amazon's Graviton machines, it would be very convenient if we could also run these docker images on those arm based processors. Do you have any plans to release a multiplatform version of the pact-cli?

**Comment by bethesque** [comment]
Yup. I'd started work in one of the other docker repos (the [pact broker](https://github.com/pact-foundation/pact-broker-docker/pull/72/files) one) but haven't had time to finish it. Would you be interested in submitting a PR? Happy to support you with any ruby related things if you can address the docker build requirements.

**Comment by YOU54F** [comment]
We currently wouldn't be able to test as there isn't m1 support in GH actions. 

see https://github.com/actions/runner/issues/805

I've noted this package, so running a self hosted GH actions.

https://github.com/mas-cli/m1-github-actions-runner

@csydvs would you interested in submitting a PR? As said, we would be able to help support and I know this would be a benefit to good few

**Comment by csydvs** [comment]
Hey guys, not a docker build expert and not going to be top of my priority, but I can try and pull my weight if I get a chance. Just in the mean time, if someone else volunteers, they're more than welcome to it.

@YOU54F are you saying that you're already running a self-hosted runner, or that you'd want that to be somehow part of the PR? I only have passing familiarity with Github Actions, would like to know what your vision is for this?

**Comment by ireinhart** [comment]
Any progress, anyone?

Would nice to see the pactfoundation/pact-cli image as an arm64 version.

**Comment by YOU54F** [comment]
look literally just above where you’ve typed, there are linked pull reqs

**Comment by YOU54F** [comment]
# TL;DR

> You can now get multi-platform images by appending `-multi` to your `pact-cli` Docker tag

### ARM64

```console
❯docker run --rm --platform=linux/arm64 -it pactfoundation/pact-cli:latest-multi /bin/sh -c 'uname -sm'
Unable to find image 'pactfoundation/pact-cli:latest-multi' locally
latest-multi: Pulling from pactfoundation/pact-cli
Digest: sha256:d0f8f7d47796cc33c5c47ef328bb1ba42685d4c7005dece026ea8363f9d3ce65
Status: Downloaded newer image for pactfoundation/pact-cli:latest-multi
Linux aarch64
```

### AMD64

```console
~ on ☁️  (eu-west-2) took 7s
❯ docker run --rm --platform=linux/amd64 -it pactfoundation/pact-cli:latest-multi /bin/sh -c 'uname -sm'
Unable to find image 'pactfoundation/pact-cli:latest-multi' locally
latest-multi: Pulling from pactfoundation/pact-cli
Digest: sha256:d0f8f7d47796cc33c5c47ef328bb1ba42685d4c7005dece026ea8363f9d3ce65
Status: Downloaded newer image for pactfoundation/pact-cli:latest-multi
Linux x86_64
```

### ARM

```console
~ on ☁️  (eu-west-2) took 6s
❯ docker run --rm --platform=linux/arm -it pactfoundation/pact-cli:latest-multi /bin/sh -c 'uname -sm'
Unable to find image 'pactfoundation/pact-cli:latest-multi' locally
latest-multi: Pulling from pactfoundation/pact-cli
Digest: sha256:d0f8f7d47796cc33c5c47ef328bb1ba42685d4c7005dece026ea8363f9d3ce65
Status: Downloaded newer image for pactfoundation/pact-cli:latest-multi
Linux armv7l
```
